src/multitaper.jl

```diff
@@ -0,0 +1,518 @@
+    plan = onesided ? plan_rfft(fft_input_tmp; flags=fft_flags) : plan_fft(fft_input_tmp; flags=fft_flags)
+    if window === nothing
+        r = fs * ntapers
+        window = dpss(n_samples, nw, ntapers)
+    else
+        r = fs * sum(abs2, window)
+    end
+
+    return MTConfig{T}(n_samples, nfft, n_freq_bins, ntapers, freq, fs, plan,
+                       fft_input_tmp, fft_output_tmp, window, onesided, r)
+end
+
+allocate_output(config::MTConfig{T}) where T = Vector{fftabs2type(T)}(undef, config.n_freq_bins)
+
+"""
+    tapered_spectra!(output, signal::AbstractVector, config) where {T}
+
+Internal function used in [`mt_pgram!`](@ref) and [`mt_cross_spectral!`](@ref).
+
+* `output`: `length(frequencies)` x `ntapers`
+* `signal`: `n_samples`
+* `config.fft_input_tmp`: `nfft`
+* `config.window`: `n_samples` x `ntapers`
+* `config.plan`: 1D plan for `nfft`
+
+"""
+@views function tapered_spectra!(output, signal::AbstractVector, config::MTConfig)
+    if length(signal) != config.n_samples
+        throw(DimensionMismatch("Expected `length(signal) == config.n_samples`; got `length(signal)` = $(length(signal)) and `config.n_samples` = $(config.n_samples)"))
+    end
+    if size(output) != (config.n_freq_bins, config.ntapers)
+        throw(DimensionMismatch("Expected `size(output) == (config.n_freq_bins, config.ntapers)`; got `size(output)` = $(size(output)) and `(config.n_freq_bins, config.ntapers)` = $((config.n_freq_bins, config.ntapers))"))
+
+    end
+    input = config.fft_input_tmp
+    
+    @inbounds for j in 1:(config.ntapers)
+        for i in 1:length(signal)
+            input[i] = config.window[i, j] * signal[i]
+        end
+        # zero-pad
+        input[(length(signal) + 1):(config.nfft)] .= 0
+
+        mul!(output[:, j], config.plan, input)
+    end
+    return nothing
+end
+
+"""
+    mt_pgram!(output::AbstractVector, signal::AbstractVector, config::MTConfig) -> Periodogram
+
+Computes a multitapered periodogram with parameters specifed by `config`,
+storing the output in `output`.
+
+* `signal::AbstractVector`: should be of length `config.n_samples`
+* `output::AbstractVector`: should be of length `config.n_freq_bins`
+"""
+function mt_pgram!(output::AbstractVector, signal::AbstractVector, config::MTConfig)
+    if length(output) != config.n_freq_bins
+        throw(DimensionMismatch("""Expected `output` to be of length `config.n_freq_bins`;
+                got `length(output) = $(length(output)) and `config.n_freq_bins` = $(config.n_freq_bins)"""))
+    end
+    if length(signal) != config.n_samples
+        throw(DimensionMismatch("""Expected `signal` to be of length `config.n_samples`;
+                got `length(signal) == $(length(signal)) and `config.n_samples` = $(config.n_samples)"""))
+    end
+    tmp = config.fft_output_tmp
+    tapered_spectra!(tmp, signal, config)
+
+    output .= 0
+    for j in 1:(config.ntapers)
+        fft2pow!(output, @view(tmp[:, j]), config.nfft, config.r, config.onesided)
```

Comment:
here, we make use that `fft2pow!` increments in-place, so calling it repeatedly adds the contribution over each taper